ibm/resource_ibm_kms_key_policy.go

```diff
@@ -0,0 +1,476 @@
+							Computed:    true,
+							Description: "Updates when the policy is replaced or modified. The date format follows RFC 3339.",
+						},
+						"enabled": {
+							Type:        schema.TypeBool,
+							Required:    true,
+							Description: "If set to true, Key Protect enables a dual authorization policy on a single key.",
+						},
+					},
+				},
+			},
+			ResourceName: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The name of the resource",
+			},
+			ResourceCRN: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The crn of the resource",
+			},
+			ResourceStatus: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The status of the resource",
+			},
+		},
+	}
+}
+func resourceIBMKmsKeyPolicyCreate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	kpAPI, err := meta.(ClientSession).keyManagementAPI()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	rContollerClient, err := meta.(ClientSession).ResourceControllerAPIV2()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	log.Println("Entering the Policies++++++++>>>>>>>>")
+
+	instanceID := d.Get("instance_id").(string)
+	endpointType := d.Get("endpoint_type").(string)
+	key_id := d.Get("key_id").(string)
+
+	rContollerApi := rContollerClient.ResourceServiceInstanceV2()
+
+	instanceData, err := rContollerApi.GetInstance(instanceID)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	instanceCRN := instanceData.Crn.String()
+	crnData := strings.Split(instanceCRN, ":")
+	var hpcsEndpointURL string
+
+	if crnData[4] == "hs-crypto" {
+		hpcsEndpointAPI, err := meta.(ClientSession).HpcsEndpointAPI()
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		resp, err := hpcsEndpointAPI.Endpoint().GetAPIEndpoint(instanceID)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		if endpointType == "public" {
+			hpcsEndpointURL = "https://" + resp.Kms.Public + "/api/v2/keys"
+		} else {
+			hpcsEndpointURL = "https://" + resp.Kms.Private + "/api/v2/keys"
+		}
+
+		u, err := url.Parse(hpcsEndpointURL)
+		if err != nil {
+			return diag.Errorf("Error Parsing hpcs EndpointURL")
+		}
+		kpAPI.URL = u
+	} else if crnData[4] == "kms" {
+		if endpointType == "private" {
+			if !strings.Contains(kpAPI.Config.BaseURL, "private") {
+				kmsEndpURL := strings.SplitAfter(kpAPI.Config.BaseURL, "https://")
+				if len(kmsEndpURL) == 2 {
+					kmsEndpointURL := kmsEndpURL[0] + "private." + kmsEndpURL[1]
+					u, err := url.Parse(kmsEndpointURL)
+					if err != nil {
+						return diag.Errorf("Error Parsing kms EndpointURL")
+					}
+					kpAPI.URL = u
+				} else {
+					return diag.Errorf("Error in Kms EndPoint URL ")
+				}
+			}
+		}
+	} else {
+		return diag.Errorf("Invalid or unsupported service Instance")
+	}
+	kpAPI.Config.InstanceID = instanceID
+
+	var expiration *time.Time
+	if es, ok := d.GetOk("expiration_date"); ok {
```

Comment:
there is no `expiration_date` in schema.. How is this possible or why is this needed?